### PR TITLE
ComboboxControl: Add lint rule for 40px size prop usage

### PR DESCRIPTION
**Pull request by mirka**
Part of #63871

## What?

Add a lint rule to prevent people from introducing new usages of ComboboxControl that do not adhere to the new default size.

### Testing Instructions

The lint error should trigger for violations [as expected](https://github.com/WordPress/gutenberg/pull/64410#discussion_r1712431913).

### CODE REVIEW

File: packages/block-library/src/avatar/user-control.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ function UserControl( { value, onChange } ) {
 
 	return (
 		<ComboboxControl
+			__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/avatar/user-control.js:36]: | Before | After |
|--------|-------|
|Avatar block, before|Avatar block, after|

File: packages/block-library/src/post-author/edit.js [hunk 2]
```diff
@@ -100,74 +101,82 @@ function PostAuthorEdit( {
...
-							/>
-						) ) || (
+					<VStack
+						spacing={ 4 }
+						className="wp-block-post-author__inspector-settings"
+					>
+						{ showAuthorControl &&
+							( ( showCombobox && (
+								<ComboboxControl
+									__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/post-author/edit.js:111]: The spacing gets messed up when a ComboboxControl enters the mix, because it has an additional div wrapper outside the BaseControl. The auto margin added to BaseControls by the block inspector interprets this as a `:last-child` BaseControl, and adds a smaller margin.

So here we add a `VStack` to explicitly handle our spacing, with the correct metrics (16px gap), and additional CSS to counteract the auto margins. Painful 😭 FYI I am working on this in #64526.

| Before | After |
|--------|-------|
|Author block, before|Author block, after|

Looks like this when `showCombobox === false`:

Author block, after with select

Comment (ciampo) [reply packages/block-library/src/post-author/edit.js:111]: > Painful

:hug:

**Review by ciampo — APPROVED** [review]
🚀

File: packages/block-library/src/post-author/block.json [hunk 3]
```diff
@@ -66,5 +66,6 @@
 			"clientNavigation": true
 		}
 	},
+	"editorStyle": "wp-block-post-author-editor",
```
Comment (ciampo) [inline packages/block-library/src/post-author/block.json:69]: Why is this change needed?

Comment (tyxla) [reply packages/block-library/src/post-author/block.json:69]: It's necessary because we're readding the editor style file and we want it to be taken into consideration when previewing the blocks in the editor.

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Might be a good idea to add the prop to the component README. After all, we've already done it for `__nextHasNoMarginBottom` previously.

Should we ping some folks who previously removed the post author block editor styles just in case? See #49567 and #35513.

**Comment by mirka** [comment]
> Should we ping some folks who previously removed the post author block editor styles just in case? See https://github.com/WordPress/gutenberg/pull/49567 and https://github.com/WordPress/gutenberg/pull/35513.

Mmm, thanks for excavating that. I read through the thread and they just removed the editor styles because they realized that the existing styles were duplicated in the main `styles.scss`. In this case we are adding legitimate editor-only styles, so it's justified.
